stubs/seaborn/seaborn/_core/scales.pyi

```diff
@@ -0,0 +1,100 @@
+from _typeshed import Incomplete
+from collections.abc import Callable, Sequence
+from dataclasses import dataclass
+from typing import Any, ClassVar
+from typing_extensions import Self, TypeAlias
+
+from matplotlib.axis import Ticker
+from matplotlib.scale import ScaleBase
+from matplotlib.ticker import Formatter, Locator
+from matplotlib.units import ConversionInterface
+from numpy.typing import ArrayLike, NDArray
+from pandas import Series
+from seaborn._core.typing import Default
+
+TransFuncs: TypeAlias = tuple[Callable[[ArrayLike], ArrayLike], Callable[[ArrayLike], ArrayLike]]
+Pipeline: TypeAlias = Sequence[Callable[[Any], Any] | None]
+
+class Scale:
+    values: tuple[Incomplete] | str | list[Incomplete] | dict[Incomplete, Incomplete] | None
+    def __post_init__(self) -> None: ...
+    def tick(self) -> Self: ...
+    def label(self) -> Self: ...
+    def __call__(self, data: Series[Any]) -> ArrayLike: ...
+
+@dataclass
+class Boolean(Scale):
+    values: tuple[Incomplete] | list[Incomplete] | dict[Incomplete, Incomplete] | None = None
+    def tick(self, locator: Locator | None = None) -> Self: ...  # type: ignore[override]
+    def label(self, formatter: Formatter | None = None) -> Self: ...  # type: ignore[override]
+
+@dataclass
+class Nominal(Scale):
+    values: tuple[Incomplete] | str | list[Incomplete] | dict[Incomplete, Incomplete] | None = None
+    order: list[Incomplete] | None = None
+    def tick(self, locator: Locator | None = None) -> Self: ...  # type: ignore[override]
+    def label(self, formatter: Formatter | None = None) -> Self: ...  # type: ignore[override]
+
+@dataclass
+class Ordinal(Scale): ...
+
+@dataclass
+class Discrete(Scale): ...
+
+@dataclass
+class ContinuousBase(Scale):
+    values: tuple[Incomplete] | str | None = None
+    norm: tuple[Incomplete] | None = None
+
+@dataclass
+class Continuous(ContinuousBase):
+    values: tuple[Incomplete] | str | None = None
+    trans: str | TransFuncs | None = None
+    def tick(  # type: ignore[override]
+        self,
+        locator: Locator | None = None,
+        *,
+        at: Sequence[float] | None = None,
+        upto: int | None = None,
+        count: int | None = None,
+        every: float | None = None,
+        between: tuple[float, float] | None = None,
+        minor: int | None = None,
+    ) -> Self: ...
+    def label(  # type: ignore[override]
+        self,
+        formatter: Formatter | None = None,
+        *,
+        like: str | Callable[[float], str] | None = None,
+        base: int | None | Default = ...,
+        unit: str | None = None,
+    ) -> Self: ...
+
+@dataclass
+class Temporal(ContinuousBase):
+    trans: ClassVar[Incomplete]  # not sure it is a classvar but the runtime has no annotation so it is not a dataclass field
```

Comment:
Seems like it's just unused? Might be better to omit it.

Reply:
It is used in the base class. It is also defined as a dataclass field in `Continuous` which is a sibling class to `Temporal`. So I'd say it must exist but it is weirdly defined.